src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1422,15 +1431,25 @@ internal void ClearTransaction()
 
         /// <summary>
         /// Creates another connector and sends a cancel request through it for this connector.
+        /// Returns, whether cancel request was send.
         /// </summary>
-        internal void CancelRequest(bool throwExceptions = false)
+        internal bool CancelRequest(bool throwExceptions = false, bool requestedByUser = true)
```

Comment:
It's pretty odd that if an IO exception occurred, this will return true (unless throwExceptions is true).

nit: rename to TryCancelRequest (because returns bool)